src/components/external/PaymentLinkDetails/components/PaymentLinkDetails/PaymentLinkDetails.tsx

```diff
@@ -0,0 +1,139 @@
+import cx from 'classnames';
+import useCoreContext from '../../../../../core/Context/useCoreContext';
+import { useState, useMemo, useEffect, useCallback } from 'preact/hooks';
+import ButtonActions from '../../../../../components/internal/Button/ButtonActions/ButtonActions';
+import { ButtonVariant } from '../../../../internal/Button/types';
+import { useConfigContext } from '../../../../../core/ConfigContext';
+import { useFetch } from '../../../../../hooks/useFetch';
+import { EMPTY_OBJECT } from '../../../../../utils';
+import Header from '../../../../../components/internal/Header';
+import { PaymentLinkDetailsProps } from '../../types';
+import { ExternalUIComponentProps } from 'src/components/types';
+import { useModalContext } from '../../../../internal/Modal/Modal';
+import { PaymentLinkSummary } from '../PaymentLinkSummary/PaymentLinkSummary';
+import './PaymentLinkDetails.scss';
+import { PaymentLinkExpiration } from '../PaymentLinkExpiration/PaymentLinkExpiration';
+import { PaymentLinkTabs } from '../PaymentLinkTabs/PaymentLinkTabs';
+import { ButtonActionsList } from '../../../../../components/internal/Button/ButtonActions/types';
+
+const CLASSNAMES = {
+    root: 'adyen-pe-payment-link-details',
+    content: 'adyen-pe-payment-link-details__content',
+};
+
+export const PaymentLinkDetails = ({ id, onUpdate, hideTitle }: ExternalUIComponentProps<PaymentLinkDetailsProps>) => {
+    const { i18n } = useCoreContext();
+    const { getPayByLinkPaymentLinkById } = useConfigContext().endpoints;
+    const {
+        data: paymentLink,
+        isFetching,
+        error,
+        refetch,
+    } = useFetch(
+        useMemo(
+            () => ({
+                fetchOptions: {
+                    enabled: !!id && !!getPayByLinkPaymentLinkById,
+                },
+                queryFn: async () => {
+                    return getPayByLinkPaymentLinkById!(EMPTY_OBJECT, {
+                        path: {
+                            paymentLinkId: id,
+                        },
+                    });
+                },
+            }),
+            [getPayByLinkPaymentLinkById, id]
+        )
+    );
+
+    const [activeScreen, setActiveScreen] = useState<'details' | 'expirationConfirmation'>('details');
+    const [isCopiedIndicatorVisible, setCopiedIndicatorVisible] = useState(false);
+
+    useEffect(() => {
```

Comment:
This effect can be replaced by moving the timeout directly into the event handler:

```
const handleCopyLink = useCallback(async () => {
    if (!paymentLink) return;

    try {
        await navigator.clipboard.writeText(paymentLink.linkInformation.paymentLink);
        setCopiedIndicatorVisible(true);
        setTimeout(() => setCopiedIndicatorVisible(false), 2000);
    } catch (error) {
        console.error('Failed to copy link:', error);
    }
}, [paymentLink]);
```

Since the state change only originates from `handleCopyLink`, we can handle the reset there directly instead of syncing through an [effect](https://react.dev/learn/you-might-not-need-an-effect)